src/timepicker/timepicker.ts

```diff
@@ -248,6 +249,53 @@ export class NgbTimepicker implements ControlValueAccessor,
     this.propagateModelChange();
   }
 
+  mouseHourEvent(event: MouseWheelEvent) {
+    if (!this.disabled) {
+      let step: number = this.isScrollUp(event) ? this.hourStep : -this.hourStep;
+      this.changeHour(step);
+    }
+    event.preventDefault();
+  }
+
+  mouseMinuteEvent(event: MouseWheelEvent) {
+    if (!this.disabled) {
+      let step: number = this.isScrollUp(event) ? this.minuteStep : -this.minuteStep;
+      this.changeMinute(step);
+    }
+    event.preventDefault();
+  }
+
+  mouseSecondEvent(event: MouseWheelEvent) {
+    if (!this.disabled) {
+      let step: number = this.isScrollUp(event) ? this.minuteStep : -this.minuteStep;
+      this.changeSecond(step);
+    }
+    event.preventDefault();
+  }
+
+  keyDownHour(event: KeyboardEvent) {
+    if (this.validEvent(event)) {
+      this.isArrowUp(event) ? this.changeHour(this.hourStep) : this.changeHour(-this.hourStep);
```

Comment:
One thing that might reduce the repetition & logic here is to make use of Angular's helpers for binding to event listeners, i.e. `keydown.ArrowUp` and `keydown.ArrowDown` - this will allow you to avoid needing to do a check for whether it is a valid event.